packages/x-components/src/x-modules/queries-preview/components/query-preview-list.vue

```diff
@@ -0,0 +1,215 @@
+<template>
+  <ul>
+    <li v-for="query in renderedQueries" :key="query" data-test="query-preview-item">
+      <QueryPreview @load="flagAsLoaded" @error="flagAsFailed" v-bind="$attrs" :query="query">
+        <template v-for="(_, slotName) in $scopedSlots" v-slot:[slotName]="scope">
+          <slot :name="slotName" v-bind="scope" />
+        </template>
+      </QueryPreview>
+    </li>
+  </ul>
+</template>
+
+<script lang="ts">
+  import Vue from 'vue';
+  import { Component, Prop, Watch } from 'vue-property-decorator';
+  import StaggeredFadeAndSlide from '../../../components/animations/staggered-fade-and-slide.vue';
+  import { xComponentMixin } from '../../../components/x-component.mixin';
+  import { RequestStatus } from '../../../store';
+  import { queriesPreviewXModule } from '../x-module';
+  import QueryPreview from './query-preview.vue';
+
+  interface QueryPreviewStatusRecord {
+    [query: string]: RequestStatus;
+  }
+
+  /**
+   * Renders the results previews of a list of queries, and exposes the {@link QueryPreview} slots
+   * to modify the content.
+   * The requests are made in sequential order.
+   *
+   * @public
+   */
+  @Component({
+    inheritAttrs: false,
+    components: {
+      QueryPreview,
+      StaggeredFadeAndSlide
+    },
+    mixins: [xComponentMixin(queriesPreviewXModule)]
+  })
+  export default class QueryPreviewList extends Vue {
+    /**
+     * List of queries to preview.
+     */
+    @Prop({ required: true })
+    public queries!: string[];
+
+    /**
+     * Contains the status of the preview requests, indexed by query.
+     */
+    public queriesStatus: QueryPreviewStatusRecord = {};
+
+    /**
+     * Gets all the queries to render, that are those that don't have an `error` status.
+     *
+     * @returns A list of queries.
+     * @internal
+     */
+    protected get renderedQueries(): string[] {
+      return this.queries.filter(
+        query => this.queriesStatus[query] === 'success' || this.queriesStatus[query] === 'loading'
+      );
+    }
+
+    /**
+     * Resets the status of all queries if they change.
+     *
+     * @internal
+     */
+    @Watch('queries', { immediate: true })
+    protected resetStatusRecord(): void {
```

Comment:
I was trying to follow the approach that we used in the `base-result-image` where instead of mapping the status of the current query, we save a copy of the queries as pendingQueries and every time that a query is being load or set as failed, we remove it from the pendindQueries or add it to the loadedQueries, but i couldnt make it works. I am not sure if should be a better approach or not